masq_lib/src/utils.rs

```diff
@@ -109,6 +115,18 @@ pub fn exit_process(code: i32, message: &str) {
     }
 }
 
+#[cfg(not(target_os = "windows"))]
+pub fn exit_process_with_sigterm(message: &str) {
+    if unsafe { RUNNING_TEST } {
+        panic!("{}", message);
+    } else {
+        eprintln!("{}", message);
+        not_win_cfg::signal::raise(not_win_cfg::signal::SIGTERM).expect("sigterm failure");
+        //This function must not return; so wait for death.
+        std::thread::sleep(not_win_cfg::Duration::from_millis(300))
```

Comment:
Maybe this should be longer, in case it takes the SIGTERM handler longer to close everything out. Could be days and months and years, for that matter: it should never complete, ever.